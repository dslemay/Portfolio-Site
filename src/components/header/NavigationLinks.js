import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { css } from 'emotion';
import Link from '../ScrollLink';

const LinksContainer = styled.div`
  margin-bottom: 0;
  margin-right: 0;
  background-color: ${props => props.theme.baseColor};

  a {
    color: ${props => props.theme.lightAccent};
    cursor: pointer;
    transition: color 500ms;

    &:visited {
      color: ${props => props.theme.lightAccent};
    }

    &:hover {
      color: ${props => props.theme.complementaryDark};
    }
  }
`;

const MobileLinks = css`
  position: absolute;
  right: 0px;
  padding: 0 1em;
  padding-top: 0.5em;
  transition: all 800ms;
  z-index: 9;

  a {
    margin-bottom: 0.75em;
  }
`;

const MobileLinksOpen = css`
  ${MobileLinks};
  top: 63px;
  opacity: 1;
`;

const MobileLinksClosed = css`
  ${MobileLinks};
  top: -100px;
  opacity: 0;
`;

const LinkStyleDesktop = css`
  margin-right: 1.5em;
  text-decoration: none;
  display: inline;
`;

const LinkStyleMobile = css`
  margin-right: 0;
  text-decoration: none;
  display: block;
`;

const NavLink = props => (
  <Link
    to={props.to}
    href={`#${props.to}`} // Needed for accessibility of react-scroll implementation
    className={props.mobile ? LinkStyleMobile : LinkStyleDesktop}
    smooth
    offset={-80}
    duration={1000}
  >
    {props.text}
  </Link>
);

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  mobile: PropTypes.bool,
  text: PropTypes.string.isRequired
};

NavLink.defaultProps = {
  mobile: false
};

function renderLinksLayout(mobile, isOpen) {
  if (!mobile) return '';
  return mobile && isOpen ? MobileLinksOpen : MobileLinksClosed;
}

const NavigationLinks = props => (
  <LinksContainer className={renderLinksLayout(props.mobile, props.isOpen)}>
    <NavLink to="about" text="About" mobile={props.mobile} />
    <NavLink to="work" text="Work" mobile={props.mobile} />
    <NavLink to="contact" text="Contact" mobile={props.mobile} />
    <a href="/blog" className={props.mobile ? LinkStyleMobile : LinkStyleDesktop}>
      Blog
    </a>
  </LinksContainer>
);

NavigationLinks.propTypes = {
  mobile: PropTypes.bool,
  isOpen: PropTypes.bool
};

NavigationLinks.defaultProps = {
  mobile: false,
  isOpen: false
};

export default NavigationLinks;
