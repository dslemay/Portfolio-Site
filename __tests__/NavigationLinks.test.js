import React from 'react';
import renderer from 'react-test-renderer';
import NavigationLinks from '../src/components/header/NavigationLinks';

describe('NavigationControls', () => {
  it('renders properly', () => {
    const tree = renderer.create(<NavigationLinks home />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
