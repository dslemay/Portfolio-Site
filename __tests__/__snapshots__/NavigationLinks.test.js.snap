// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationControls renders properly 1`] = `
<div
  className="css-gdfr0c css-11ymm0i0"
>
  <MockedLink
    className="css-14xxsjd"
    to="#about"
  >
    About
  </MockedLink>
  <MockedLink
    className="css-14xxsjd"
    to="#work"
  >
    Work
  </MockedLink>
  <MockedLink
    className="css-14xxsjd"
    to="/contact"
  >
    Contact
  </MockedLink>
  <MockedLink
    className="css-14xxsjd"
    to="/blog"
  >
    Blog
  </MockedLink>
</div>
`;
